src/sentry/snuba/discover.py

```diff
@@ -2,17 +2,66 @@
 
 import six
 
+from collections import namedtuple
 from copy import deepcopy
-from sentry.api.event_search import get_filter, resolve_field_list
+from datetime import datetime
+
+from sentry.api.event_search import (
+    TAG_KEY_RE,
+    get_filter,
+    resolve_field_list,
+    find_reference_event,
+    InvalidSearchQuery,
+)
+from sentry.snuba.events import get_columns_from_aliases
 from sentry.utils.snuba import (
     Dataset,
+    SnubaTSResult,
     DISCOVER_COLUMN_MAP,
     QUOTED_LITERAL_RE,
     get_function_index,
     raw_query,
     transform_results,
+    zerofill,
 )
 
+ReferenceEvent = namedtuple("ReferenceEvent", ["organization", "slug", "fields"])
+
+
+def create_reference_event_conditions(reference):
+    """
+    Create a list of conditions based on a Reference object.
+
+    This is useful when you want to get results that match an exemplar
+    event. A use case of this is generating pagination links for, or getting
+    timeseries results of the records inside a single aggregated row.
+
+    reference (Reference) The reference event to build conditions from.
+    """
+    columns = get_columns_from_aliases(reference.fields)
+    event_data = find_reference_event(reference.organization, reference.slug, columns)
+
+    conditions = []
+    tags = {}
+    if "tags.key" in event_data and "tags.value" in event_data:
```

Comment:
What causes "tags.key" and "tags.value" to be present in event_data. It seems to be coming from what's in the groupby clause, but why would we ever group by "tags.key" or "tags.value"?

Reply:
I see. Should we just add the individual tags we want instead of the entire tag lists? (I feel like I might've noticed some timeouts on these). Eventstore can provide a way to do this.

Reply:
I think that would be a good improvement that we can do.